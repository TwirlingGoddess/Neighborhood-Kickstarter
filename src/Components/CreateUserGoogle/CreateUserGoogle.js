import React, { Component } from 'react';
import Select from 'react-select';
import GoogleLogin from 'react-google-login';

import { getNeighborhoods, addNewUser } from '../../utilities/apiCalls/apiCalls'

import './CreateUserGoogle.css'

class CreateUserGoogle extends Component {
  constructor() {
    super()
    this.state = {
      userName: '',
      neighborhood: '',
      neighborhoods: []
    }
  }

  componentDidMount = () => {
    this.setNeighborhoods();
  }

  responseGoogle = async (response) => {
    let { updateUser } = this.props;
    let googleSignIn = {
      first_name: response.profileObj.givenName,
      last_name: response.profileObj.familyName,
      email: response.profileObj.email,
      username: this.state.userName,
      district_id: this.state.neighborhood.id,
      token: response.profileObj.googleId
    };
    
    const user = await addNewUser(googleSignIn);

    if (user) {
      updateUser(user);
      this.props.history.push('/Landing')
    }
  }

  setNeighborhoods = async() => {
    const response = await getNeighborhoods();

    const neighborhoods = response.map(neighborhood => {
      return { value: neighborhood.name, label: neighborhood.name, id: neighborhood.id }
    })

    this.setState({
      neighborhoods
    })
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    })
  }

  handleSelectChange = (selectedOption) => {
    const neighborhood = selectedOption;
    this.setState({
      neighborhood
    })
  }

  handleSubmit = (event) => {
    event.preventDefault();

    this.setState({
      userName: '',
      neighborhood: ''
    })

    this.props.history.push('/Landing')
  }

  render() {
    return (
      <div className='create-user-section'>
        <form className='create-user-form-google' onSubmit={this.handleSubmit}>
          <h1>Create Account</h1>
          <input
            className='input-fields-google'
            name='userName'
            type='text'
            value={this.state.userName}
            placeholder='User Name'
            onChange={this.handleChange}
          />
          <Select
            className='select-input' 
            placeholder='Select your neighborhood'
            value={this.state.neighborhood}
            onChange={this.handleSelectChange}
            options={this.state.neighborhoods}
          />
          <GoogleLogin
            clientId="603748791729-1qgv1pg7tl426jut42re2tnub34nu0hu.apps.googleusercontent.com"
            buttonText="Login with Google"
            onClick={this.responseGoogle}
            disabled={this.state.neighborhood && this.state.userName ? false : true}
            onSuccess={this.responseGoogle}
            onFailure={this.responseGoogle}
          />
        </form>
      </div>
    )
  }
}

export default CreateUserGoogle