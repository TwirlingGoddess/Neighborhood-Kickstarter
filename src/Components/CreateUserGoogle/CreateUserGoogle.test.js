import React from 'react';
import CreateUserGoogle from './CreateUserGoogle';
import { shallow } from 'enzyme';

describe('CreateUserGoogle', () => {
  let wrapper;
  it('should match the snapshot', () => {
    wrapper = shallow(<CreateUserGoogle />);
    expect(wrapper).toMatchSnapshot();
  });
}); 