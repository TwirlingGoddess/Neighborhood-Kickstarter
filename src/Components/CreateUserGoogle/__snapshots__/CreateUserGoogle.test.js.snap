// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateUserGoogle should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateUserGoogle />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="create-user-form-google"
        onSubmit={[Function]}
      >
        <h1>
          Create Account
        </h1>
        <input
          className="input-fields-google"
          name="userName"
          onChange={[Function]}
          placeholder="User Name"
          type="text"
          value=""
        />
        <StateManager
          className="select-input"
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          onChange={[Function]}
          options={Array []}
          placeholder="Select your neighborhood"
          value=""
        />
        <t
          accessType="online"
          buttonText="Login with Google"
          clientId="603748791729-1qgv1pg7tl426jut42re2tnub34nu0hu.apps.googleusercontent.com"
          cookiePolicy="single_host_origin"
          disabled={true}
          disabledStyle={
            Object {
              "opacity": 0.6,
            }
          }
          fetchBasicProfile={true}
          isSignedIn={false}
          jsSrc="https://apis.google.com/js/client:platform.js"
          onClick={[Function]}
          onFailure={[Function]}
          onRequest={[Function]}
          onSuccess={[Function]}
          prompt=""
          scope="profile email"
          tag="button"
          type="button"
          uxMode="popup"
        />
      </form>,
      "className": "create-user-section",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Create Account
          </h1>,
          <input
            className="input-fields-google"
            name="userName"
            onChange={[Function]}
            placeholder="User Name"
            type="text"
            value=""
          />,
          <StateManager
            className="select-input"
            defaultInputValue=""
            defaultMenuIsOpen={false}
            defaultValue={null}
            onChange={[Function]}
            options={Array []}
            placeholder="Select your neighborhood"
            value=""
          />,
          <t
            accessType="online"
            buttonText="Login with Google"
            clientId="603748791729-1qgv1pg7tl426jut42re2tnub34nu0hu.apps.googleusercontent.com"
            cookiePolicy="single_host_origin"
            disabled={true}
            disabledStyle={
              Object {
                "opacity": 0.6,
              }
            }
            fetchBasicProfile={true}
            isSignedIn={false}
            jsSrc="https://apis.google.com/js/client:platform.js"
            onClick={[Function]}
            onFailure={[Function]}
            onRequest={[Function]}
            onSuccess={[Function]}
            prompt=""
            scope="profile email"
            tag="button"
            type="button"
            uxMode="popup"
          />,
        ],
        "className": "create-user-form-google",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Create Account",
          },
          "ref": null,
          "rendered": "Create Account",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-fields-google",
            "name": "userName",
            "onChange": [Function],
            "placeholder": "User Name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "select-input",
            "defaultInputValue": "",
            "defaultMenuIsOpen": false,
            "defaultValue": null,
            "onChange": [Function],
            "options": Array [],
            "placeholder": "Select your neighborhood",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessType": "online",
            "buttonText": "Login with Google",
            "clientId": "603748791729-1qgv1pg7tl426jut42re2tnub34nu0hu.apps.googleusercontent.com",
            "cookiePolicy": "single_host_origin",
            "disabled": true,
            "disabledStyle": Object {
              "opacity": 0.6,
            },
            "fetchBasicProfile": true,
            "isSignedIn": false,
            "jsSrc": "https://apis.google.com/js/client:platform.js",
            "onClick": [Function],
            "onFailure": [Function],
            "onRequest": [Function],
            "onSuccess": [Function],
            "prompt": "",
            "scope": "profile email",
            "tag": "button",
            "type": "button",
            "uxMode": "popup",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="create-user-form-google"
          onSubmit={[Function]}
        >
          <h1>
            Create Account
          </h1>
          <input
            className="input-fields-google"
            name="userName"
            onChange={[Function]}
            placeholder="User Name"
            type="text"
            value=""
          />
          <StateManager
            className="select-input"
            defaultInputValue=""
            defaultMenuIsOpen={false}
            defaultValue={null}
            onChange={[Function]}
            options={Array []}
            placeholder="Select your neighborhood"
            value=""
          />
          <t
            accessType="online"
            buttonText="Login with Google"
            clientId="603748791729-1qgv1pg7tl426jut42re2tnub34nu0hu.apps.googleusercontent.com"
            cookiePolicy="single_host_origin"
            disabled={true}
            disabledStyle={
              Object {
                "opacity": 0.6,
              }
            }
            fetchBasicProfile={true}
            isSignedIn={false}
            jsSrc="https://apis.google.com/js/client:platform.js"
            onClick={[Function]}
            onFailure={[Function]}
            onRequest={[Function]}
            onSuccess={[Function]}
            prompt=""
            scope="profile email"
            tag="button"
            type="button"
            uxMode="popup"
          />
        </form>,
        "className": "create-user-section",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Create Account
            </h1>,
            <input
              className="input-fields-google"
              name="userName"
              onChange={[Function]}
              placeholder="User Name"
              type="text"
              value=""
            />,
            <StateManager
              className="select-input"
              defaultInputValue=""
              defaultMenuIsOpen={false}
              defaultValue={null}
              onChange={[Function]}
              options={Array []}
              placeholder="Select your neighborhood"
              value=""
            />,
            <t
              accessType="online"
              buttonText="Login with Google"
              clientId="603748791729-1qgv1pg7tl426jut42re2tnub34nu0hu.apps.googleusercontent.com"
              cookiePolicy="single_host_origin"
              disabled={true}
              disabledStyle={
                Object {
                  "opacity": 0.6,
                }
              }
              fetchBasicProfile={true}
              isSignedIn={false}
              jsSrc="https://apis.google.com/js/client:platform.js"
              onClick={[Function]}
              onFailure={[Function]}
              onRequest={[Function]}
              onSuccess={[Function]}
              prompt=""
              scope="profile email"
              tag="button"
              type="button"
              uxMode="popup"
            />,
          ],
          "className": "create-user-form-google",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Create Account",
            },
            "ref": null,
            "rendered": "Create Account",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-fields-google",
              "name": "userName",
              "onChange": [Function],
              "placeholder": "User Name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "select-input",
              "defaultInputValue": "",
              "defaultMenuIsOpen": false,
              "defaultValue": null,
              "onChange": [Function],
              "options": Array [],
              "placeholder": "Select your neighborhood",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessType": "online",
              "buttonText": "Login with Google",
              "clientId": "603748791729-1qgv1pg7tl426jut42re2tnub34nu0hu.apps.googleusercontent.com",
              "cookiePolicy": "single_host_origin",
              "disabled": true,
              "disabledStyle": Object {
                "opacity": 0.6,
              },
              "fetchBasicProfile": true,
              "isSignedIn": false,
              "jsSrc": "https://apis.google.com/js/client:platform.js",
              "onClick": [Function],
              "onFailure": [Function],
              "onRequest": [Function],
              "onSuccess": [Function],
              "prompt": "",
              "scope": "profile email",
              "tag": "button",
              "type": "button",
              "uxMode": "popup",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
